arelle/XbrlConst.py

```diff
@@ -1,122 +1,151 @@
-from arelle.ModelValue import qname
+from __future__ import annotations
 import os
-try:
-    from regex import compile as re_compile
-except ImportError:
-    from re import compile as re_compile
+from typing import TYPE_CHECKING
+from typing import cast
+from typing import Tuple  # tuple type conflicts with xbrl tuple namespace
+from regex import compile as re_compile
+from arelle.ModelValue import qname
+
+if TYPE_CHECKING:
+    from regex import Pattern
+    from arelle.ModelValue import QName
+    from arelle.ModelObject import ModelObject
+    from arelle.typing import TypeGetText
+
+    _: TypeGetText
 
 xsd = "http://www.w3.org/2001/XMLSchema"
-qnXsdSchema = qname("{http://www.w3.org/2001/XMLSchema}xsd:schema")
-qnXsdAppinfo = qname("{http://www.w3.org/2001/XMLSchema}xsd:appinfo")
-qnXsdDefaultType = qname("{http://www.w3.org/2001/XMLSchema}xsd:anyType")
+qnXsdSchema: QName = qname("{http://www.w3.org/2001/XMLSchema}xsd:schema")
```

Comment:
Thanks to the overloads, can these explicit QName types be removed now?
```suggestion
qnXsdSchema = qname("{http://www.w3.org/2001/XMLSchema}xsd:schema")
```